plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/part/AttributePart.java

```diff
@@ -69,6 +75,58 @@ public String getAttributeLabel() {
         return ERDUIUtils.getFullAttributeLabel(getDiagram(), getAttribute(), false);
     }
 
+    @Override
+    protected void addSourceConnection(ConnectionEditPart connection, int index) {
+        final DBPPreferenceStore store = ERDUIActivator.getDefault().getPreferences();
+        if (!store.getString(ERDUIConstants.PREF_ROUTING_TYPE).equals(ERDUIConstants.ROUTING_MIKAMI) || ERDAttributeVisibility.isHideAttributeAssociations(store)) {
+            return;
+        }
+        if (((AssociationPart) connection).getAssociation().getSourceAttributes().contains(getAttribute())) {
+            super.addSourceConnection(connection, index);
+        }
+    }
+
+    @Override
+    protected List<ERDAssociation> getModelSourceConnections() {
+        final DBPPreferenceStore store = ERDUIActivator.getDefault().getPreferences();
+        if (!store.getString(ERDUIConstants.PREF_ROUTING_TYPE).equals(ERDUIConstants.ROUTING_MIKAMI) || ERDAttributeVisibility.isHideAttributeAssociations(store)) {
+            return Collections.emptyList();
+        }
+        List<ERDAssociation> list = new ArrayList<>();
+        for (ERDAssociation erdAssociation : super.getModelSourceConnections()) {
+            if (erdAssociation.getSourceAttributes().contains(getAttribute()) && erdAssociation.getSourceEntity() != null) {
+                list.add(erdAssociation);
+            }
+        }
+        return list;
+    }
+
+    @Override
+    protected List<ERDAssociation> getModelTargetConnections() {
+        final DBPPreferenceStore store = ERDUIActivator.getDefault().getPreferences();
+        if (!store.getString(ERDUIConstants.PREF_ROUTING_TYPE).equals(ERDUIConstants.ROUTING_MIKAMI) || ERDAttributeVisibility.isHideAttributeAssociations(store)) {
+            return Collections.emptyList();
+        }
+        List<ERDAssociation> list = new ArrayList<>();
+        for (ERDAssociation erdAssociation : super.getModelTargetConnections()) {
+            if (erdAssociation.getTargetAttributes().contains(getAttribute()) && erdAssociation.getTargetEntity() != null) {
+                list.add(erdAssociation);
+            }
+        }
+        return list;
+    }
+
+    @Override
+    protected void addTargetConnection(ConnectionEditPart connection, int index) {
+        final DBPPreferenceStore store = ERDUIActivator.getDefault().getPreferences();
+        if (!store.getString(ERDUIConstants.PREF_ROUTING_TYPE).equals(ERDUIConstants.ROUTING_MIKAMI) || ERDAttributeVisibility.isHideAttributeAssociations(store)) {
+            return;
+        }
+        if (((AssociationPart) connection).getAssociation().getTargetAttributes().contains(getAttribute())) {
+            super.addTargetConnection(connection, index);
+        }
+    }
```

Comment:
I don't like that it's hardcoded just for specific router.

Can we abstract these things into a dedicated "Decorator" class?